demo/csi/nfs/volume.hcl

```
@@ -0,0 +1,9 @@
+id        = "csi-nfs"
+name      = "csi-nfs"
+type      = "csi"
+plugin_id = "rocketduck-nfs"
+
+capability {
+  access_mode     = "multi-node-multi-writer"
+  attachment_mode = "file-system"
+}
```

CSI's capabilities don't enclose each other the way you'd expect. The first job that claims the volume sets the active capability from the set of capabilities in the volume spec. Having a single `capability` means that the volume _cannot_ be mounted in a non-multi-node-multi-writer way by jobs. If your application doesn't safely support multi-writer semantics or you never want the volume to be writable, you'll probably want to be able to tell Nomad to lock that out. 

So I'd expand this to at least cover:

```suggestion
capability {
  access_mode     = "multi-node-multi-writer"
  attachment_mode = "file-system"
}

capability {
  access_mode     = "multi-node-single-writer"
  attachment_mode = "file-system"
}

capability {
  access_mode     = "multi-node-reader-only"
  attachment_mode = "file-system"
}
```

---

Something new to learn every day, thanks!